backend/src/main/java/fk/prof/backend/model/association/BackendDetail.java

```diff
@@ -0,0 +1,104 @@
+package fk.prof.backend.model.association;
+
+import recording.Recorder;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.Set;
+
+public class BackendDetail {
+  private final String backendIPAddress;
+  private final int reportingFrequencyInSeconds;
+  private final int maxAllowedSkips;
+  private final Set<Recorder.ProcessGroup> associatedProcessGroups;
+
+  private Double lastReportedLoad = null;
+  //Last reported time is initialized with epochSecond=0 to ensure isDefunct returns true until backend reports its load to the leader
+  private LocalDateTime lastReportedTime = LocalDateTime.ofEpochSecond(0, 0, ZoneOffset.UTC);
+
+  public BackendDetail(String backendIPAddress, int reportingFrequencyInSeconds, int maxAllowedSkips)
+      throws IOException {
+    this(backendIPAddress, reportingFrequencyInSeconds, maxAllowedSkips, null);
+  }
+
+  public BackendDetail(String backendIPAddress, int reportingFrequencyInSeconds, int maxAllowedSkips, byte[] processGroupsBytes)
+      throws IOException {
+    this.backendIPAddress = backendIPAddress;
+    this.reportingFrequencyInSeconds = reportingFrequencyInSeconds;
+    this.maxAllowedSkips = maxAllowedSkips;
+
+    Recorder.ProcessGroups processGroups = processGroupsBytes == null
+        ? Recorder.ProcessGroups.newBuilder().build()
+        : Recorder.ProcessGroups.parseFrom(processGroupsBytes);
+    if(processGroups != null) {
+      this.associatedProcessGroups = new HashSet<>(processGroups.getProcessGroupList());
+    } else {
+      this.associatedProcessGroups = new HashSet<>();
+    }
+  }
+
+  public void reportLoad(double loadFactor) {
+    this.lastReportedLoad = loadFactor;
+    this.lastReportedTime = LocalDateTime.now(Clock.systemUTC());
```

Comment:
Using system nanotime with updateAndGet semantics to do updates for latest clocked report only